Add missing AUTOINC/SERIAL for FAB tables

ashb opened the pull request:
In 1.10.13 we introduced a migration that creates the tables with the server_default but that migration only did anything if the tables didn't already exist. But the tables created by the FAB model have a default (but not a server_default).

Oh, and the final bit of the puzzle, in 2.4 we finally "took control" of the FAB security models in to airflow and those do not have the default set.

Fixes #26497

ashb committed:
Add missing AUTOINC/SERIAL for FAB tables

In 1.10.13 we introduced a migration that creates the tables with the
server_default but that migration only did anything if the tables didn't
already exist. But the tables created by the FAB model have a default
(but not a server_default).

Oh, and the final bit of the puzzle, in 2.4 we finally "took control" of
the FAB security models in to airflow and those do not have the default
set.

kaxil commented on airflow/migrations/versions/0118_2_4_2_add_missing_autoinc_fab.py:
```diff
@@ -0,0 +1,69 @@
+    """Apply add_missing_autoinc_fab"""
+    conn = op.get_bind()
+    if conn.dialect.name in ['mssql', 'sqlite']:
+        # 1.10.12 didn't support SQL Server, so it couldn't have gottren this wrong --> nothing to correct
```
```suggestion
        # 1.10.12 didn't support SQL Server, so it couldn't have gotten this wrong --> nothing to correct
```

ashb replied:
```suggestion
    """
    Apply migration.
    
    If these columns are already of the right type (i.e. created by our migration in 1.10.13 rather than FAB
    itself in an earlier version), then this migration will issue an alter statement to change them to what
    they already -- i.e. its a no-op.
    
    These tables are small (100 to low 1k rows at most) so even if it did try to change it it's not costly to
    do.
    """
    conn = op.get_bind()
    if conn.dialect.name in ['mssql', 'sqlite']:
        # 1.10.12 didn't support SQL Server, so it couldn't have gotten this wrong --> nothing to correct
```

ashb committed:
Update airflow/migrations/versions/0118_2_4_2_add_missing_autoinc_fab.py

jedcunningham commented on airflow/migrations/versions/0118_2_4_2_add_missing_autoinc_fab.py:
```diff
@@ -0,0 +1,78 @@
+                kwargs['type_'] = sa.Sequence(f'{table}_id_seq').next_value()
+            else:
+                kwargs['autoincrement'] = True
+            batch.alter_column("id", existing_type=sa.Integer(), eixsintg_nullable=False, **kwargs)
```
```suggestion
            batch.alter_column("id", existing_type=sa.Integer(), existing_nullable=False, **kwargs)
```

blag commented on airflow/migrations/versions/0118_2_4_2_add_missing_autoinc_fab.py:
```diff
@@ -0,0 +1,78 @@
+    
+    If these columns are already of the right type (i.e. created by our migration in 1.10.13 rather than FAB
+    itself in an earlier version), then this migration will issue an alter statement to change them to what
+    they already -- i.e. its a no-op.
```
```suggestion
    they already are -- i.e. its a no-op.
```

uranusjr committed:
Fix static checks

ashb committed:
Run migrations with with a pool of a connection.

Without this `create_session()` will open a new connection, and that causes mysql
to hang waiting to get a "metadata lock on table".

Using the "stock" pool with size=1 and max_overflow=0 doesn't work, that
instead times out if you try to get a new connection from the pool.
SingletonThreadPool instead returns the existing active connection which
is what we want.
